### PR TITLE
Embed SonarC# 8.48 and SonarVB 8.48

### LINKED ISSUE #3330: Embed SonarC# 8.49 and SonarVB 8.49
In release 8.49 there were mostly FPs and FNs fixed caused by the new C# 11 syntax, and also 1 AD0001 bug was fixed.

Release notes: https://github.com/SonarSource/sonar-dotnet/releases/tag/8.49.0.57237

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #3330

### CODE REVIEW

File: src/Roslyn.Suppressions/Roslyn.Suppressions/SupportedSuppressionBuilder.g.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ private static ImmutableArray<SuppressionDescriptor> GetDescriptors()
                 // If the number of diagnostic ids changes significantly or decreases when the analysers are updated,
                 // investigate! There may be a problem with code generator.
                 // *************************************************************************************************
-                // Number of unique diagnostic ids (C# and VB.NET): 429
+                // Number of unique diagnostic ids (C# and VB.NET): 435
```
Comment (duncanp-sonar) [inline src/Roslyn.Suppressions/Roslyn.Suppressions/SupportedSuppressionBuilder.g.cs:51]: Whenever we update the Sonar Roslyn analyzers, we should re-run the `.tt` file that generates the list of issues that can be suppressed.

I've re-run it manually this time, and added #3335 to the backlog.

File: src/EmbeddedSonarAnalyzer.props [hunk 2]
```diff
@@ -5,7 +5,7 @@
   <PropertyGroup>
     <!-- Note: don't forget to upload the new packages to the package feed when changing 
          the SonarAnalyzer version -->
-    <EmbeddedSonarAnalyzerVersion>8.48.0.56517</EmbeddedSonarAnalyzerVersion>
+    <EmbeddedSonarAnalyzerVersion>8.49.0.57237</EmbeddedSonarAnalyzerVersion>
```
Comment (duncanp-sonar) [inline src/EmbeddedSonarAnalyzer.props:8]: The LT DotNET released a new version of their analyzers today.

**Review by ugras-ergun-sonarsource — APPROVED** [review]

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]
Just a question: it says that the build pipeline has failed, why did that happen?

**Comment by duncanp-sonar** [comment]
@georgii-borovinskikh-sonarsource you can follow the links from the "checks" to the [pipeline run](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=57342&view=results) to see the issues.

The first issue isn't one I've seen before - some problem on the build agent restoring NuGet packages.
The second is a known file-locking issue.

The normal approach is to re-run the failed jobs.

**Comment by georgii-borovinskikh-sonarsource** [comment]
@duncanp-sonar Yeah I saw the nuget one and got confused a bit

**Pull request merged**